src/DeviceInterfaceValues.tsx

```diff
@@ -184,6 +189,346 @@ const InterfaceData = ({ interfaceData }: InterfaceDataProps): React.ReactElemen
+          .typeError('The value must be integer.');
+      case 'double':
+        return yup
+          .number()
+          .required('This field is required and must be a number.')
+          .typeError('The value must be number.');
+      case 'boolean':
+        return yup
+          .boolean()
+          .required('This field is required and must be true or false.')
+          .typeError('The value must be either true or false.');
+      case 'string':
+        return yup
+          .string()
+          .required('This field is required and must be a string.')
+          .typeError('The value must be a valid string.');
+      case 'binaryblob':
+        return yup
+          .string()
+          .matches(base64Regex, 'The value must be a valid binary blob encoded.')
+          .required('This field is required and must be a binary blob.');
+      case 'datetime':
+        return yup
+          .string()
+          .matches(iso8601Regex, 'The date must be in the format "YYYY-MM-DDTHH:mm:ss.sssZ".')
+          .required('This field is required and must be a valid datetime.');
+      case 'doublearray':
+        return yup
+          .array()
+          .of(yup.number())
+          .required('This field is required and must be an array of numbers.')
+          .typeError('Each item in the array must be a valid number.');
+      case 'integerarray':
+      case 'longintegerarray':
+        return yup
+          .array()
+          .of(yup.number())
+          .required('This field is required and must be an array of integers.')
+          .typeError('Each item in the array must be a valid integer.');
+      case 'booleanarray':
+        return yup
+          .array()
+          .of(yup.boolean())
+          .required('This field is required and must be an array of booleans.')
+          .typeError('Each item in the array must be either true or false.');
+      case 'stringarray':
+        return yup
+          .array()
+          .of(yup.string())
+          .required('This field is required and must be an array of strings.')
+          .typeError('Each item in the array must be a valid string.');
+      case 'binaryblobarray':
+        return yup
+          .array()
+          .of(yup.string().matches(base64Regex, 'Each item must be a valid binary blob encoded.'))
+          .required('This field is required and must be an array of binary blobs.')
+          .typeError('Each item in the array must be a valid binary blob.');
+      case 'datetimearray':
+        return yup
+          .array()
+          .of(
+            yup
+              .string()
+              .matches(iso8601Regex, 'Each item must be in the format "YYYY-MM-DDTHH:mm:ss.sssZ".'),
+          )
+          .required('This field is required and must be an array of datetime strings.')
+          .typeError('Each item in the array must be a valid datetime string.');
+      default:
+        return yup.string().required('This field is required.');
+    }
+  };
```

Comment:
This is a generic concept that does not depend on other data from this component. We should define it outside of the render function. Even better, we could define a `InterfaceValue` model here https://github.com/astarte-platform/astarte-dashboard/tree/master/src/astarte-client/models and use that to define all the utility functions needed, simplifying the code of this page by a lot